pallets/market/src/lib.rs

```diff
@@ -1002,15 +834,15 @@ pub mod pallet {
         ) -> Result<Cid, DispatchError> {
             let pieces = proposals
                 .map(|p| {
-                    let cid = p.cid()?;
-                    let commitment =
-                        Commitment::from_cid(&cid).map_err(|err| ProposalError::CommD(err))?;
+                    let cid = p.cid().map_err(|_| PaddedPieceSizeError::InvalidPieceCid)?;
+                    let commitment = Commitment::from_cid(&cid)
+                        .map_err(|_| PaddedPieceSizeError::UnableToCreateCommD)?;
```

Comment:
This is actually not a CommD, it's a Piece CID aka. CommP (it's on a deal, multiple CommPs == CommD)

Reply:
Good catch! Updated in the latest commit